python/mdvtools/llm/langchain_mdv.py

```
@@ -289,92 +195,165 @@ def agent_with_contextualization(question):
-            # shortly after this...
-            # Error in StdOutCallbackHandler.on_chain_start callback: AttributeError("'NoneType' object has no attribute 'get'")
-            self.log(f"Asking the LLM: '{question}'")
+            log(f"Asking the LLM: '{question}'")
             if self.init_error:
-                # let's think about doing something better here.
-                self.socket_api.update_chat_progress(
+                socket_api.update_chat_progress(
                     f"Agent initialisation error: {str(self.init_error)}", id, 100, 0
                 )
                 return f"The agent had an error during initialisation: \n\n{str(self.init_error)[:500]}"
             full_prompt = prompt_data + "\nQuestion: " + question
```

Flake8 (7.2.0)

[error] local variable 'full_prompt' is assigned to but never used

(F841)